plugins/woocommerce/src/Admin/API/Init.php

```diff
@@ -54,115 +55,167 @@ public function __construct() {
-					'Automattic\WooCommerce\Admin\API\Leaderboards',
-					'Automattic\WooCommerce\Admin\API\Reports\Controller',
-					'Automattic\WooCommerce\Admin\API\Reports\Import\Controller',
-					'Automattic\WooCommerce\Admin\API\Reports\Export\Controller',
-					'Automattic\WooCommerce\Admin\API\Reports\Products\Controller',
-					'Automattic\WooCommerce\Admin\API\Reports\Variations\Controller',
-					'Automattic\WooCommerce\Admin\API\Reports\Products\Stats\Controller',
-					'Automattic\WooCommerce\Admin\API\Reports\Variations\Stats\Controller',
-					'Automattic\WooCommerce\Admin\API\Reports\Revenue\Stats\Controller',
-					'Automattic\WooCommerce\Admin\API\Reports\Orders\Controller',
-					'Automattic\WooCommerce\Admin\API\Reports\Orders\Stats\Controller',
-					'Automattic\WooCommerce\Admin\API\Reports\Categories\Controller',
-					'Automattic\WooCommerce\Admin\API\Reports\Taxes\Controller',
-					'Automattic\WooCommerce\Admin\API\Reports\Taxes\Stats\Controller',
-					'Automattic\WooCommerce\Admin\API\Reports\Coupons\Controller',
-					'Automattic\WooCommerce\Admin\API\Reports\Coupons\Stats\Controller',
-					'Automattic\WooCommerce\Admin\API\Reports\Stock\Controller',
-					'Automattic\WooCommerce\Admin\API\Reports\Stock\Stats\Controller',
-					'Automattic\WooCommerce\Admin\API\Reports\Downloads\Controller',
-					'Automattic\WooCommerce\Admin\API\Reports\Downloads\Stats\Controller',
-					'Automattic\WooCommerce\Admin\API\Reports\Customers\Controller',
-					'Automattic\WooCommerce\Admin\API\Reports\Customers\Stats\Controller',
-				);
-
-				// The performance indicators controllerq must be registered last, after other /stats endpoints have been registered.
-				$analytics_controllers[] = 'Automattic\WooCommerce\Admin\API\Reports\PerformanceIndicators\Controller';
-
-				$analytics_controllers = array_merge( $analytics_controllers, $analytic_mu_controllers );
-			}
-
-			$controllers = array_merge( $controllers, $analytics_controllers, $analytic_mu_controllers );
+		if ( ! did_filter( 'woocommerce_admin_rest_controllers' ) ) {
+			/**
+			 * Filter for the WooCommerce Admin REST controllers.
+			 *
+			 * @param array $controllers List of rest API controllers.
+			 *
+			 * @since 3.5.0
+			 */
+			$controllers = apply_filters( 'woocommerce_admin_rest_controllers', $controllers );
+		}
+
+		foreach ( $controllers as $controller ) {
+			$this->$controller = new $controller();
+			$this->$controller->register_routes();
 		}
 
 		/**
-		 * Filter for the WooCommerce Admin REST controllers.
+		 * Fires immediately after the WooCommerce core `wc-admin` routes have been registered.
 		 *
-		 * @since 3.5.0
-		 * @param array $controllers List of rest API controllers.
+		 * @since 10.3.0
 		 */
-		$controllers = apply_filters( 'woocommerce_admin_rest_controllers', $controllers );
+		do_action( 'woocommerce_wc-admin_rest_namespace_routes_registered' ); // phpcs:ignore WordPress.NamingConventions.ValidHookName.UseUnderscores
```

Comment:
Do we have to break naming conventions? We can sanitise `-` to `_` or pass as prop as an alternative.